agent/nginx_reload_manager.py

```diff
@@ -0,0 +1,213 @@
+                to_delete = paths[:-1]
+                for path in to_delete:
+                    with contextlib.suppress(FileNotFoundError):
+                        # In case RQ worker removed the file
+                        # before we could delete it
+                        os.remove(path)
+                    self.log(f"Deleted {path}", print_always=True)
+                return "Skipped"  # Send skipped status to retry the job
+
+            raise Exception(f"Not able to auto fix the issue for domain : {domain!s}") from e
+        except Exception as e:
+            import traceback
+
+            traceback.print_exc()
+            self.log(f"Error while reloading nginx : {e!s}")
+            return "Failure"
+
+    def _should_skip_nginx_reload(self):
+        """
+        ├─983331 nginx: worker process is shutting down
+        ├─983332 nginx: worker process is shutting down
+        ├─983333 nginx: worker process is shutting down
+        ├─983334 nginx: worker process is shutting down
+        ├─983361 nginx: worker process
+        ├─983362 nginx: worker process
+        ├─983363 nginx: worker process
+        ├─983364 nginx: worker process
+        └─983369 nginx: cache manager process
+        """
+        try:
+            status = subprocess.run(
+                "sudo systemctl status nginx", capture_output=True, shell=True
+            ).stdout.decode("utf-8")
+            total_workers = status.count("nginx: worker process")
+            dying_workers = status.count("nginx: worker process is shutting down")
+            active_workers = max(1, total_workers - dying_workers)
+            return (dying_workers / active_workers) >= 2
+        except Exception as e:
+            self.log(f"Failed to check nginx status : {e!s}")
+            return False
+
+    def _dequeue_jobs(self) -> list[str]:
+        # Fetch up to 10000 jobs, which might not be hit ever
+        job_ids = self.redis.lrange(PENDING_QUEUE, 0, self.batch_size)
+        if job_ids:
+            # Trim pending queue
+            self.redis.ltrim(PENDING_QUEUE, len(job_ids), -1)
+            # Move jobs to processing queue
+            self.redis.rpush(PROCESSING_QUEUE, *job_ids)
+
+        # Fetch job IDs from processing queue
+        # Important to fetch from processing queue to ensure than if worker died previously,
+        # old jobs in processing queue are still processed
+        return self.redis.lrange(PROCESSING_QUEUE, 0, self.batch_size)
+
+    def _update_status_and_cleanup(self, job_ids: list[str], status: str):
+        for job_id in job_ids:
+            self.redis.set(RELOAD_REQUEST_STATUS_FORMAT.format(job_id), status)
+        self.redis.ltrim(PROCESSING_QUEUE, len(job_ids), -1)
+
+    def log(self, message, print_always=False):
+        if not self.debug and not print_always:
+            return
+        print(f"[{datetime.now()}] {message!s}")
+
+    def exit_gracefully(self, signum, frame):
+        self.exit_requested = True
```

Comment:
Add forced shutdown on retry. Sometimes our graceful logic can get stuck and never restart on SIGTERMs.

Reply:
Can be just done by checking here if `self.exit_requested` is already true.